### PR TITLE
Add a new command: do-compare-based-on-last-two-clipboard-items.

### PR DESCRIPTION
**Pull request by zw963**

### CODE REVIEW

File: Global/README.md [hunk 1]
```diff
@@ -81,3 +81,7 @@ E.g. changes "Do androids dream of electric sheep?" to "Do Androids Dream of Ele
 
 Toggles between upper- and lower-case letters in selected text.
 
+### [Run your's preferred compare tools based on the last two items on clipboard history](do-compare-based-on-last-two-clipboard-items.ini)
```
Comment (hluk) [inline Global/README.md]: The title should be something short here so it's easier to refer to it later with a shorter more permanent link. E.g.: "Compare Last Two Clipboard History Items"

File: Global/README.md [hunk 2]
```diff
@@ -81,3 +81,7 @@ E.g. changes "Do androids dream of electric sheep?" to "Do Androids Dream of Ele
...
+
+This is just a sample command, in fact, you can run any command your choice(`bcompare in this script`), only if this command accept the last
```
Comment (hluk) [inline Global/README.md]: Can you link to where to get the `bcompare`?

**Comment by zw963** [comment]
Done, please check, thank you.

**Comment by hluk** [comment]
Thanks!

**Comment by pbodnar** [comment]
Oh, what a coincidence, I was just about to do the same stuff, thanks for contributing this! :) I've just come here when searching for an inspiration of how to write to a `TemporaryFile` before passing it to [WinMerge](https://manual.winmerge.org/en/Command_line.html), for a change.

Some remarks for a potential discussion:

1. I would suggest to rename the ini file: shouldn't it use "-" for consistency with the other files?
2. I always wonder why names selected for various commands need to differ so much - what do you think about using at least similar names at the level of the README.md and ini files? It would help users to quickly decide which commands they have already installed and which not.
3. There are some unnecessary typos in the command description. Apart from others, I would probably omit mentioning the global key for keeping it simple, as well as for consistency with the other commands' descriptions - suggested keyboard shortcuts are part of ini files already.

Also, but that is quite for another ticket, I think it would be great to combine the [Replace in Selection command](https://github.com/hluk/copyq-commands/blob/master/Global/replace-all-occurences-in-selected-text.ini) with this one, so that a user can immediately see what gets replaced. (BTW It would be also nice to have user defined replacements, similar to how the [Snippets command](https://github.com/hluk/copyq-commands/blob/master/Global/snippets.ini) work.)

**Comment by pbodnar** [comment]
For anyone interested, the following is possibly an ideal options set passed to the aforementioned WinMerge executable. The command's script line which passes the temporary files to the diff tool:

```js
execute('winmergeu', '/e', '/x', '/u', '/fl', '/dl', 'item1', '/dr', 'item2', name1, name2)
```

* `/e` ... close with escape
* `/x` ... only show a message dialog when items are identical and exit
* `/u` ... don't add comparison to the MRU list
* `/fl` ... put focus to the left editor (the 1st item)
* `/dl` and `/dr` ... titles for the editors

**Comment by zw963** [comment]
@pbodnar , sorry for name issue, i am not a english native speaker, can you please create a PR for fix my name issue? i think @hluk  will very luck to merge your work.

**Pull request merged**